test/functional/lua/fs_spec.lua

```diff
@@ -366,6 +366,24 @@ describe('vim.fs', function()
       )
     end)
 
+    it('assings priorities according to nested markers', function()
+      local bufnr = api.nvim_get_current_buf()
+      eq(
+        vim.fs.joinpath(test_source_path, 'test/functional'),
+        exec_lua(
+          [[return vim.fs.root(..., { 'example_spec.lua', {'CMakeLists.txt', 'CMakePresets.json'}, '.luarc.json'})]],
+          bufnr
+        )
+      )
+      eq(
+        vim.fs.joinpath(test_source_path, 'test/functional/fixtures'),
+        exec_lua(
+          [[return vim.fs.root(..., { {'CMakeLists.txt', 'CMakePresets.json'}, 'example_spec.lua', '.luarc.json'})]],
```

Comment:
do we need a test for the heterogenuos case?

    [[return vim.fs.root(..., { function() ... end, 'example_spec.lua', '.luarc.json'})]],